src/components/HabitForm.tsx

```diff
@@ -2,17 +2,54 @@ import moment from "moment";
 import React, { useState } from "react";
 import useForm from "react-hook-form";
 import DatePicker from "react-datepicker";
-import { editHabit, addHabit } from "../actions";
-import { connect } from "react-redux";
+import { editHabit, addHabit } from "../store/actions";
+import { connect, ConnectedProps } from "react-redux";
 import "react-datepicker/dist/react-datepicker.css";
 import { updateHabitStreak } from "../utils/streak";
+import { State } from "../store/types";
+import { RouteComponentProps } from "react-router-dom";
 
-const Form = props => {
+const mapState = (state: State, ownProps: OwnProps) => {
+  const user = state.currentUser;
+  if (user && ownProps.id) {
+    return {
+      habit: state.userStates[user].habits[+ownProps.id],
```

Comment:
Should we also verify we can parse `id` into a number here (as we need to in the `HabitCard`)? Could probably use the same solution in both places.